### ISSUE TITLE
Freezing while drawing with pencil tool

### ISSUE THREAD

**Issue by sebba-dev** [aseprite/aseprite#2789]
So I don't know if I'm doing something wrong here but Aseprite is the only application that does this.

Everything works normally but what looks like to me at random times when drawing with pencil tool Aseprite freezes for about 500ms and then continues like nothing happened. There are no errors or warns in the logs and debug logs.

The intervals for freezing look random to me, 5-30 seconds. If I knew C++ a bit better I would try to fix the problem myself but I'm kinda lost in the source code right now.

I tried fixes like UI scaling and changing cursors, cross hair type, brush preview, etc.. nothing fixes the problem

Used CXX flags: `-march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection`

TLDR; When drawing Aseprite freezes for approximately 500ms and than works normally again this repeats every 5-30 seconds

### Aseprite and System version

* Aseprite version: v1.2.27-dev
* System: Arch Linux, nouvenau drivers, 5.12.12-arch1-1 kernel

**Comment by dacap** [comment]
Hi @SebbaIndustries, can you check if this happens in the [v1.3-beta4 version](https://www.aseprite.org/beta/)?

There are [some changes](https://github.com/aseprite/aseprite/issues/2089#issuecomment-855106019) in that branch that might help in your situation, but I'm not 100% (the changes are not yet public in this repository).

**Comment by SebbaIndustries** [comment]
So I tried to v1.3-beta4-x64 version but the problem is still present, I didn't really notice any difference between versions.

**Comment by SebbaIndustries** [comment]
So I did some profiling with `perf` [here is 30 second dump](https://op65n.org/shared/aseprite_dump.json), freeze happened at around 10 seconds. You can use [speedscope](https://www.speedscope.app/) to get a flame graph.

I know this is not that relevant, but I tried libresprite and it runs fine without freezing. Maybe this can help to point out the issue?

**Comment by dacap** [comment]
Thanks for the info @SebbaIndustries, just in case, as you were able to compile Aseprite before, could you give a try to commenting out [this if()](https://github.com/aseprite/laf/blob/92e2e5b9e5f4813d33577fd06ff5d92e2a481053/os/x11/x11.h#L37) (located in `laf/os/x11/x11.h`), changing the code from:
```c++
    m_display = XOpenDisplay(nullptr);
    m_xim = XOpenIM(m_display, nullptr, nullptr, nullptr);
    if (m_display)
      m_xinput.load(m_display);
  }
```
to
```c++
    m_display = XOpenDisplay(nullptr);
    m_xim = XOpenIM(m_display, nullptr, nullptr, nullptr);
    //if (m_display)
    //  m_xinput.load(m_display);
  }
```

**Comment by SebbaIndustries** [comment]
When I comment pit these two lines I get a runtime error: `fish: Job 1, './aseprite' terminated by signal SIGSEGV (Address boundary error)`
You probably also need a stack trace for this, but I compiled Aseprite again using `-DCMAKE_BUILD_TYPE=Debug` option and got the same error. Is there a different flag to enable stack traces?

**Comment by dacap** [comment]
Thanks for giving a try. I thought that disabling the xinput would be easier, but maybe you keep those lines as they were, and try commenting [this one](https://github.com/aseprite/laf/blob/92e2e5b9e5f4813d33577fd06ff5d92e2a481053/os/x11/window.cpp#L147).
```c++
  // Receive stylus/eraser events
  //X11::instance()->xinput().selectExtensionEvents(m_display, m_window);
```
Not sure if that would work, but I'm trying to find out if this issue is related with Xinput.

**Comment by SebbaIndustries** [comment]
No errors this time, but freezing is still present. I also think there could be an issue with some Xorg component.

**Comment by SebbaIndustries** [comment]
So, I started using official Nvidia drivers instead of nouveau and the freezing stopped.. I have no idea why this is the case as Aseprite was the only application that had this problem but this somehow fixed it.

I think you can close the issue now.
